feature/amazon/src/main/java/com/revenuecat/purchases/amazon/ProxyAmazonBillingActivityBroadcastReceiver.kt

```diff
@@ -0,0 +1,38 @@
+package com.revenuecat.purchases.amazon
+
+import android.app.Activity
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import androidx.annotation.VisibleForTesting
+import java.lang.ref.WeakReference
+
+internal class ProxyAmazonBillingActivityBroadcastReceiver(activity: Activity) : BroadcastReceiver() {
+
+    private val activity: WeakReference<Activity>
+
+    init {
+        this.activity = WeakReference(activity)
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    var onReceiveCalled = false
```

Comment:
So this is only added for a test... I'm not sure but is there a way of testing whether the activity was finished in the test scenario? If so we could remove this property entirely. If not, I guess we can keep it for better tests.

Reply:
I tried but I couldn't find anything 😢 I thought about checking if state is `ON_DESTROY` but it wasn't passing because `sendBroadcast` is asynchronous. Maybe I can mock that?

Reply:
It will probably be hard to mock, since it's created by the activity. I think it's ok, we can leave as is 👍